apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -837,25 +891,43 @@ def test(element_html: str, data: pl.ElementTestData) -> None:
     # TODO: The only wrong answer being tested is the correct answer with the first
     # block mising. We should instead do a random selection of correct and incorrect blocks.
     elif data["test_type"] == "incorrect":
-        answer = filter_multiple_from_array(
-            data["correct_answers"][answer_name], ["inner_html", "indent", "uuid"]
-        )
+        answer = deepcopy(data["correct_answers"][answer_name])
+        distractor_feedback = {
+            item["uuid"]: item["distractor_feedback"]
+            for item in data["params"][answer_name]
+            if not item["is_correct"]
+        }
+
         answer.pop(0)
         score = 0
         if grading_method is GradingMethodType.UNORDERED or (
             (
-                grading_method is GradingMethodType.DAG
-                or grading_method is GradingMethodType.RANKING
+                grading_method in [GradingMethodType.DAG, GradingMethodType.RANKING]
+                and partial_credit_type is PartialCreditType.LCS
             )
-            and partial_credit_type is PartialCreditType.LCS
         ):
             score = round(float(len(answer)) / (len(answer) + 1), 2)
 
         if (
-            grading_method is GradingMethodType.DAG
-            and feedback_type is FeedbackType.FIRST_WRONG
+            grading_method
+            in [
+                GradingMethodType.DAG,
+                GradingMethodType.RANKING,
+            ]
+            and feedback_type in FIRST_WRONG_TYPES
         ):
-            feedback = FIRST_WRONG_FEEDBACK["wrong-at-block"].format(1)
+            first_wrong = 0
```

Comment:
This logic seems very similar to some feedback setting logic above. I know this is test code, but having that factored out into a function would be cleaner IMO.

Reply:
done.